docs/design-proposals/multi-tenancy/multi-tenancy-network-mizar-integration.md

```diff
@@ -0,0 +1,201 @@
+   &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3.3.3 [Workflow for secondary subnet in VPC](#pod-creation-second-subnet-wf)<br>
+
+## 1. Introduction <a name="intro"></a>
+
+This document is a part of the multi-tenancy network design document that focuses on how to integrated [Mizar network solution](https://github.com/CentaurusInfra/mizar) into Arktos.
+
+The generic network design is in [Multi-Tenancy Network](multi-tenancy-network.md).
+
+## 2. Goals <a name="goal"></a>
+
+* Primary: support multi-tenancy communication based on existing Mizar network implementation. Multi-tenancy communication 
+includes connectivity amongs pods in the same tenant, as well as network isolation across different tenants.
+* Primary: upon tenant creation, automatically create Mizar VPC and subnet for future communication within this tenant.
+* Primary: upon tenant pod creation, automatically attach pod to Mizar VPC/subnet created for this tenant.
+* Primary: Mizar and Arktos decouple, i.e. Mizar manage VPCs and subnets in VPC only, it should not be aware of the ownership
+of different pods and other objects.
+* Secondary: support multiple subnets in same VPC for a single tenant. (Post 2022-01-30)
+* Secondary: support multiple VPC/subnets in same tenant. (Post 2022-01-30)
+
+## 3. Proposal <a name="proposal"></a>
+
+Inject Mizar adapter into Arktos, completely isolated from Mizar management plane and relatively isolated from Arktos network
+management components.
+
+### 3.1. Related Mizar Specs <a name="mizar-spec"></a>
+
+* VPC Spec
+```
+apiVersion: "mizar.com/v1"
+kind: Vpc
+metadata:
+  name: <vpc name>
+spec:
+  ip: "<ip address>"
+  prefix: "<ip address prefix length>"
+  dividers: <number of dividers in vpc>
+  status: "<vpc provisioning status>"
+```
+
+* Subnet Spec
+```
+apiVersion: mizar.com/v1
+kind: Subnet
+metadata:
+  name: <subnet name>
+spec:
+  ip: "<ip address in vpc>"
+  prefix: "<ip address prefix length>"
+  bouncers: <number of bouncers in subnet>
+  vpc: "<vpc name>"
+  status: "<subnet provisioning status>"
+```
+
+* Pod Spec
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: <pod name>
+  annotations:
+    mizar.com/vpc: "<vpc name>"
+    mizar.com/subnet: "<subnet name>"
+```
+
+### 3.2 Related Arktos Specs <a name="arktos-spec"></a>
+* Network Spec
+```
+apiVersion: arktos.futurewei.com/v1
+kind: Network
+metadata:
+  name: <VPC name defined in arktos>
+  tenant: <tenant name>
+spec:
+  type: vpc
+  vpcID: <VPC name in Mizar>
+```
+
+* Pod Spec
+```
+piVersion: v1
+kind: Pod
+metadata:
+  name: <pod name> 
+  tenant: <tenant name>
+  labels:
+    arktos.futurewei.com/network: <VPC name defined in arktos> 
+```
+
+### 3.3 Arktos Pod Creation Workflow <a name="pod-creation-wf"></a>
+#### 3.3.1 Workflow for default VPC and subnet in tenant <a name="pod-creation-default-wf"></a>
+1. Client creates Tenant
+2. Tenant controller creates default network object for tenant
```

Comment:
should be "objects" which include the default vpc object and default subnet object.

Reply:
> should be "objects" which include the default vpc object and default subnet object.

This network object means the special network object: networks.arktos.futurewei.com. And it is already implemented, not in this release.